py-polars/polars/lazyframe/frame.py

```diff
@@ -3223,7 +3223,8 @@ def group_by(
         """
         exprs = parse_as_list_of_expressions(*by, **named_by)
         lgb = self._ldf.group_by(exprs, maintain_order)
-        return LazyGroupBy(lgb)
+        lf = self
+        return LazyGroupBy(lgb, *by, _lf=lf, maintain_order=maintain_order, **named_by)
```

Comment:
Why did you change the LazyGroupBy? Seems unrelated and unnecessary.

Reply:
In the def of the lazy version of map_groups I need to start from the lf that preceded the call to `group_by` (just like in the eager one) but b/c in the lazy version it turns it into a new object in the frame context I don't have it in the LazyGroupBy context so I had to pass it through. Is there a better way to accomplish that?

Reply:
The lazy version of `map_groups` already worked correctly, right? It was just the eager version that did not support expressions?

Reply:
the lazy version doesn't raise but it also doesn't respect expressions other than taking the names. There's an example in the 2nd code block here https://github.com/pola-rs/polars/issues/13442